src/binaryen-c.cpp

```diff
@@ -58,10 +58,11 @@ BinaryenLiteral toBinaryenLiteral(Literal x) {
         case HeapType::func:
           ret.func = x.isNull() ? nullptr : x.getFunc().c_str();
           break;
-        case HeapType::any:
+        case HeapType::ext:
         case HeapType::eq:
           assert(x.isNull() && "unexpected non-null reference type literal");
           break;
+        case HeapType::any:
```

Comment:
Why is `any`'s behavior here different from that of `ext`? Is `any` not nullable?

Reply:
We only support the standardized reference types `extern` and `func` in this API and all the GC types are meant to hit that WASM_UNREACHABLE. Previously `HeapType::any` represented the standardized `extern`, but now it only represents the sepate `any` GC type.